### PR TITLE
release: Add the needed fixes for the release process

### PR DESCRIPTION
**Pull request by fidencio**
Please, check each commit for its explanation.

Fixes: #9064 -- part II

### CODE REVIEW

File: tools/packaging/release/release.sh [hunk 1]
```diff
@@ -34,6 +34,10 @@ function _check_required_env_var()
 	local env_var
 
 	case ${1} in
+		RELEASE_VERSION) env_var="${RELEASE_VERSION}" ;;
+		GH_TOKEN) env_var="${GH_TOKEN}" ;;
+		ARCHITECURE) env_var="${ARCHITECURE}" ;;
```
Comment (sprt) [inline tools/packaging/release/release.sh:39]: `s/ARCHITECURE/ARCHITECTURE`

Unless we're passed a var with a typo?

**Review by sprt — CHANGES_REQUESTED** [review]

File: tools/packaging/release/release.sh (hunk 1 above)
Comment (fidencio) [reply tools/packaging/release/release.sh]: Fixed in this not so small new series.

File: .github/workflows/release-amd64.yaml [hunk 2]
```diff
@@ -41,8 +41,10 @@ jobs:
           # We need to do such trick here as the format of the $GITHUB_REF 
           # is "refs/tags/<tag>"
           tag=$(echo $GITHUB_REF | cut -d/ -f3-)
-          tags=($tag)
-          tags+=($([[ "$tag" =~ "alpha"|"rc" ]] && echo "latest" || echo "stable"))
+          if [ "${tag}" = "main" ]; then
+              tag=$(./tools/packaging/release/release.sh next-release-version)
+          fi
+          tags=(${tag} "latest")
```
Comment (stevenhorsman) [inline .github/workflows/release-amd64.yaml:47]: Sorry for the dumb question, but is the case when we'd expect to trigger the release workflow_dispatch from a branch other than main just if are a re-creating an old release (e.g. after someone accidentally deleted it 😱). Under that circumstance do we still want the latest tag? I guess I'm trying to work out if there is a scenario that we have 3.n released and want to re-build 3.<n-1>?

**Review by stevenhorsman — APPROVED** [review]
LGTM. Just one question

File: .github/workflows/release-amd64.yaml (hunk 2 above)
Comment (fidencio) [reply .github/workflows/release-amd64.yaml:47]: That's not a dumb question, it's a very interesting one.
Releases can be created from branches, and we won't have any branch that's used apart from main.  That was my rationale when adding this logic here.

Comment (fidencio) [reply .github/workflows/release-amd64.yaml:47]: it should only occur if we're running from main, let me fix it.

Comment (fidencio) [reply .github/workflows/release-amd64.yaml]: Updated, thanks!

**Review by danmihai1 — APPROVED** [review]

**Review by GabyCT — APPROVED** [review]

**Review by sprt — APPROVED** [review]
Very nice, thank you Fabiano! Love the release notes as well.

**Pull request merged**